src/compiler/scanner.rs

```diff
@@ -6,6 +6,7 @@ struct ScanNumberReturn {
 }
 
 pub struct Scanner {
+    skip_trivia: bool,
```

Comment:
What's this do?

Reply:
Seems like it controls whether the scanner throws away whitespace vs somehow paying closer attention to it. In this instance of the scanner owned by the parser it looks like it's always set to `true` so not sure when it's in the other mode